packages/x-data-grid/src/hooks/utils/useGridApiMethod.ts

```diff
@@ -14,7 +15,7 @@ export function useGridApiMethod<
 >(privateApiRef: React.MutableRefObject<PrivateApi>, apiMethods: T, visibility: V) {
   const isFirstRender = React.useRef(true);
 
-  React.useEffect(() => {
+  useEnhancedEffect(() => {
```

Comment:
This was causing subtle bugs, because some of our hook effects use `useEnhancedEffect` (aka `React.useLayoutEffect`), so they were running before the API methods were updated, so they were calling outdated `useCallback` instances.